amethyst_tiles/src/pass.rs

```diff
@@ -0,0 +1,503 @@
+};
+
+#[cfg(feature = "profiler")]
+use thread_profiler::profile_scope;
+
+lazy_static::lazy_static! {
+    static ref TILES_VERTEX: SpirvShader = SourceShaderInfo::new(
+        include_str!("../shaders/tiles.vert"),
+        concat!(env!("CARGO_MANIFEST_DIR"), "../shaders/tiles.vert"),
+        ShaderKind::Vertex,
+        SourceLanguage::GLSL,
+        "main",
+    ).precompile().unwrap();
+
+    static ref TILES_FRAGMENT: SpirvShader = SourceShaderInfo::new(
+        include_str!("../shaders/tiles.frag"),
+        concat!(env!("CARGO_MANIFEST_DIR"), "../shaders/tiles.frag"),
+        ShaderKind::Fragment,
+        SourceLanguage::GLSL,
+        "main",
+    ).precompile().unwrap();
+
+    static ref SHADERS: ShaderSetBuilder = ShaderSetBuilder::default()
+        .with_vertex(&*TILES_VERTEX).unwrap()
+        .with_fragment(&*TILES_FRAGMENT).unwrap();
+}
+
+/// Draw opaque sprites without lighting.
+#[derive(Clone, PartialEq, Derivative)]
+#[derivative(Default(bound = ""), Debug(bound = ""))]
+pub struct DrawTiles2DDesc<T: Tile, E: CoordinateEncoder> {
+    #[derivative(Debug = "ignore")]
+    _marker: PhantomData<(T, E)>,
+}
+
+impl<T: Tile, E: CoordinateEncoder> DrawTiles2DDesc<T, E> {
+    /// Create instance of `DrawTiles2D` render group
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl<B: Backend, T: Tile, E: CoordinateEncoder> RenderGroupDesc<B, Resources>
+    for DrawTiles2DDesc<T, E>
+{
+    fn build(
+        self,
+        _ctx: &GraphContext<B>,
+        factory: &mut Factory<B>,
+        _queue: QueueId,
+        _aux: &Resources,
+        framebuffer_width: u32,
+        framebuffer_height: u32,
+        subpass: hal::pass::Subpass<'_, B>,
+        _buffers: Vec<NodeBuffer>,
+        _images: Vec<NodeImage>,
+    ) -> Result<Box<dyn RenderGroup<B, Resources>>, failure::Error> {
+        #[cfg(feature = "profiler")]
+        profile_scope!("build");
+
+        let env = FlatEnvironmentSub::new(factory)?;
+        let textures = TextureSub::new(factory)?;
+        let vertex = DynamicVertexBuffer::new();
+
+        let (pipeline, pipeline_layout) = build_tiles_pipeline(
+            factory,
+            subpass,
+            framebuffer_width,
+            framebuffer_height,
+            true,
+            vec![env.raw_layout(), textures.raw_layout()],
+        )?;
+
+        Ok(Box::new(DrawTiles2D::<B, T, E> {
+            pipeline,
+            pipeline_layout,
+            env,
+            textures,
+            vertex,
+            sprites: Default::default(),
+            _marker: PhantomData::default(),
+        }))
+    }
+}
+
+#[derive(Derivative)]
+#[derivative(Debug(bound = ""))]
+pub struct DrawTiles2D<B: Backend, T: Tile, E: CoordinateEncoder> {
+    pipeline: B::GraphicsPipeline,
+    pipeline_layout: B::PipelineLayout,
+    env: FlatEnvironmentSub<B>,
+    textures: TextureSub<B>,
+    vertex: DynamicVertexBuffer<B, SpriteArgs>,
+    sprites: OneLevelBatch<TextureId, SpriteArgs>,
```

Comment:
For transparent stuff you might want to preserve insertion order, so `OrderedOneLevelBatch` should work better.